src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/ContainingSyntaxContext.cs

```diff
@@ -75,7 +74,15 @@ public bool Equals(ContainingSyntaxContext other)
                 && ContainingNamespace == other.ContainingNamespace;
         }
 
-        public override int GetHashCode() => throw new UnreachableException();
+        public override int GetHashCode()
+        {
+            int code = ContainingNamespace?.GetHashCode() ?? 0;
+            foreach (ContainingSyntax containingSyntax in ContainingSyntax)
+            {
+                code ^= containingSyntax.Identifier.Value.GetHashCode();
+            }
+            return code;
```

Comment:
We can't use `HashCode.Combine` since we're targeting .NET Standard 2.0 (I tried to port the copy in CoreLib back to 2.0, but it's not a super clean port). We can use the `Hash` type that was ported over from Roslyn as part of Cyrus' work though.